apis/install/v1alpha1/armadaserver_types.go

```diff
@@ -0,0 +1,78 @@
+/*
+Copyright 2022.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// ArmadaServerSpec defines the desired state of ArmadaServer
+type ArmadaServerSpec struct {
+	// Name specifies the base name for all Kubernetes Resources
```

Comment:
We found out that we can remove the Name field from the Spec.  Kubernetes objects have a name field as part of the metadata.  We are using that everywhere so this field is redundant.